rskj-core/src/main/java/co/rsk/peg/bitcoin/BitcoinUtils.java

```diff
@@ -61,4 +62,26 @@ public static Optional<Script> extractRedeemScriptFromInput(TransactionInput txI
             return Optional.empty();
         }
     }
+
+    public static void removeSignaturesFromNonSegwitTransaction(BtcTransaction transaction) {
+        if (transaction.hasWitness()) {
+            throw new IllegalArgumentException("Removing signatures from SegWit transactions is not allowed.");
+        }
+
+        for (TransactionInput input : transaction.getInputs()) {
+            Script scriptSigWithoutSignatures = removeSignaturesFromInputScriptSig(input.getScriptSig());
+            input.setScriptSig(scriptSigWithoutSignatures);
+        }
+    }
+
+    private static Script removeSignaturesFromInputScriptSig(Script scriptSig) {
```

Comment:
I think this method should be renamed to a better name because it does not remove anything from the script being passed. Instead, it creates a new script from it. Some name suggestions: `createEmptyInpuScript` or `createBaseInputScript`, etc.

This is being done in [powpeg node project](https://github.com/rsksmart/powpeg-node/blob/4b730e891fd2a505a72a3afa04ffcf66a6066490/src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java#L469). Take a look, it can be a useful reference.

Reply:
I agree, but anyways, as Marcos says, there's already a `extractRedeemScriptFromInput` method in this class that can be reused. No need for this `removeSignaturesFromInputScriptSig` method.

Reply:
this method was deleted